### ISSUE TITLE
ImageClassifier with tf.data.Dataset input - classification head set loss error

### ISSUE THREAD

**Issue by qingquansong** [keras-team/autokeras#1119]
### Bug Description

The problem may be caused by the ClassificationHeadAdapter- > fit_before_convert method, which does not convert the tf TensorShape to integer. Not sure if in newer version of tf, the comparison between tf TensorShape and integer is allowed.

### Bug Reproduction
codes are in the image classifier example in jupyter (data format section)

Data used by the code:
MNIST

### Expected Behavior

![Screen Shot 2020-05-05 at 11 50 49 AM](https://user-images.githubusercontent.com/25535036/81093108-3a8a4e00-8ec7-11ea-91cf-84e7a49dcd9a.png)

### Setup Details
Include the details about the versions of:
 - OS type and version:
 - Python: 3.6
 - autokeras: 1.0.2
 - keras-tuner:
 - scikit-learn:
 - numpy:
 - pandas:
 - tensorflow: 2.1.0
